### PR TITLE
Adding Router datasource

### PR DESCRIPTION
**Pull request by maxlip**
Adding new datasource for google_compute_router

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`compute`: added `google_compute_router` datasource
```

### CODE REVIEW

**Comment by rambleraptor** [comment]
Hello! Thank you for submitting this PR. I'm going to add @slevenick on this to help you out.

As soon as you sign the CLA, we can run this through our CI system and get the review going.

**Comment by slevenick** [comment]
Hey @maxlip thanks for the contribution!

Looks pretty good! I've tested it out locally and it seems to work. It looks like the linter has [an issue with the file](https://travis-ci.org/terraform-providers/terraform-provider-google/builds/593183728), could you format it?

File: third_party/terraform/website/docs/d/datasource_compute_router.html.markdown [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+data "google_compute_router" "my-router" {
+  name   = "default-us-east1"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Optional) The name of the router. One of `name` or `self_link`
```
Comment (slevenick) [inline third_party/terraform/website/docs/d/datasource_compute_router.html.markdown]: Looks like this should be required, and not mention `self_link`

**Comment by slevenick** [comment]
Looks like there is still a conflict with `third_party/terraform/utils/provider.go.erb`.

You will need to fix that before I can generate the downstreams again

**Comment by maxlip** [comment]
Sorry I don't see a problem here that I can fix. 
I added an entry to that file as required by the addition of a new datasource, and my branch's copy it otherwise identical as far as I can see. Trying to find new ways to re-write the file just results in a blank commit over and over... 
I just don't understand what exactly to do in this case.

**Comment by slevenick** [comment]
> Sorry I don't see a problem here that I can fix.
> I added an entry to that file as required by the addition of a new datasource, and my branch's copy it otherwise identical as far as I can see. Trying to find new ways to re-write the file just results in a blank commit over and over...
> I just don't understand what exactly to do in this case.

The problem is that there is a merge conflict between your branch and the current master branch. I assume because there have been changes to that file on the master branch since you created your branch.

I would recommend pulling the current master branch and rebasing your branch on top of master. 

This guide has info on rebasing: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

The commands should look something like this:
```
git checkout master
git pull {{name-of-gcp-remote}} master
git checkout add-ds-router
git rebase master
~~ handle merge conflicts here ~~
git push {{name-of-your-remote-fork}} add-ds-router -f
```

**Comment by maxlip** [comment]
Looks like this rebase is looking better for some reason - thanks for the help and sorry to be cranky.

File: third_party/terraform/tests/data_source_google_compute_router_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceComputeRouter(t *testing.T) {
+	t.Parallel()
+	//region := getTestRegionFromEnv()
```
Comment (slevenick) [inline third_party/terraform/tests/data_source_google_compute_router_test.go]: Remove this?

File: third_party/terraform/tests/data_source_google_compute_router_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		resource "google_compute_network" "foobar" {
+			name = "%s"
+			auto_create_subnetworks = false
+		}
+		resource "google_compute_router" "foobar" {
+			name = "%s"
+			network = "${google_compute_network.foobar.name}"
+			bgp {
+				asn = 64514
+			}
```
Comment (slevenick) [inline third_party/terraform/tests/data_source_google_compute_router_test.go]: Spacing is mixed up here, can you format this like the `data "google_compute_router"` below?

**Comment by slevenick** [comment]
@maxlip Oh no something has gone wrong here.

I think that due to merging master into your branch you have picked up a bunch of unrelated changes. I would recommend either rebasing your changes onto master or creating a new branch, cherry-picking your commits from this branch and pushing that up.

**Comment by maxlip** [comment]
@slevenick looks like rebasing from scratch is worked out.

**Comment by maxlip** [comment]
OK hope that does it :)

**Pull request merged**